src/tests/functional/plugin/gpu/single_layer_tests/dynamic/top_k.cpp

```diff
@@ -0,0 +1,209 @@
+            std::iota(data.begin(), data.end(), start);
+            std::mt19937 gen(0);
+            std::shuffle(data.begin(), data.end(), gen);
+
+            if (netPrecision == ElementType::f32) {
+                auto *rawBlobDataPtr = static_cast<float *>(tensor.data());
+                for (size_t i = 0; i < size; ++i) {
+                    rawBlobDataPtr[i] = static_cast<float>(data[i]);
+                }
+            } else {
+                auto *rawBlobDataPtr = static_cast<int32_t *>(tensor.data());
+                for (size_t i = 0; i < size; ++i) {
+                    rawBlobDataPtr[i] = static_cast<int32_t>(data[i]);
+                }
+            }
+        } else if (netPrecision == ElementType::bf16) {
+            size_t O = 1, A = 1, I = 1;
+            A = shape[axis];
+            for (size_t i = 0; i < axis; i++)
+                O *= shape[i];
+            for (size_t i = axis + 1; i < shape.size(); i++)
+                I *= shape[i];
+            if (O * A * I != size)
+                FAIL() << "Incorrect blob shape " << shape;
+
+            auto *rawBlobDataPtr = static_cast<ngraph::bfloat16 *>(tensor.data());
+            for (size_t o = 0; o < O; o++) {
+                for (size_t i = 0; i < I; i++) {
+                    std::vector<int> data(A);
+                    int start = - static_cast<int>(A / 2);
+                    std::iota(data.begin(), data.end(), start);
+                    const size_t seed = (o + 1) * (i + 1);
+                    std::mt19937 gen(seed);
+                    std::shuffle(data.begin(), data.end(), gen);
+                    for (size_t a = 0; a < A; a++) {
+                        rawBlobDataPtr[o * A * I + a * I + i] = static_cast<ngraph::bfloat16>(data[a]);
+                    }
+                }
+            }
+        } else {
+            FAIL() << "generate_inputs for " << netPrecision << " precision isn't supported";
+        }
+        inputs.insert({funcInputs[0].get_node_shared_ptr(), tensor});
+    }
+
+private:
+    int64_t axis;
+    ElementType netPrecision;
+    bool staticShape;
+};
+
+TEST_P(TopKLayerGPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    run();
+}
+
+namespace {
+
+std::map<std::string, std::string> emptyAdditionalConfig;
+
+const std::vector<ElementType> netPrecisions = {
+    ElementType::f32,
+};
+
+const std::vector<int64_t> axes = {0, 1, 2, 3};
+const std::vector<int64_t> k = {1, 5, 7, 18, 21};
+
+const std::vector<ngraph::opset4::TopK::Mode> modes = {
+    ngraph::opset4::TopK::Mode::MIN,
+    ngraph::opset4::TopK::Mode::MAX
+};
+
+const std::vector<ngraph::opset4::TopK::SortType> sortTypes = {
+    ngraph::opset4::TopK::SortType::SORT_VALUES,
+    ngraph::opset4::TopK::SortType::SORT_INDICES,
+};
+
+std::vector<ov::test::InputShape> inputShapesDynamic = {
+    {{21, {20, 25}, 21, {20, 25}}, {{21, 21, 21, 21}, {21, 22, 21, 23}}}
```

Comment:
Could you add unbounded dynamic shape too?

Reply:
Added unbounded dynamic shape input param